### ISSUE TITLE
Fix S3260 FP: Should not raise for inherited generic classes

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#5160]
### Description

S3260 should not raise in this case:

### Repro steps

```csharp
    public class Outer
    {
        private class Inner { } // Noncompliant FP

        private sealed class Inner<T> : Inner { }
    }
```
